### PR TITLE
glib: Implement main context acquire guard type

### PR DESCRIPTION
**Pull request by sdroege**
Fixes https://github.com/gtk-rs/gtk-rs-core/issues/71

### CODE REVIEW

**Review by bilelmoussaoui — DISMISSED** [review]

File: glib/src/main_context.rs [hunk 1]
```diff
@@ -124,6 +124,29 @@ impl MainContext {
...
+            }
+        }
+    }
+}
+
+#[must_use = "if unused the main context will be released immediately"]
+pub struct MainContextAcquireGuard<'a>(&'a MainContext);
+
+impl<'a> Drop for MainContextAcquireGuard<'a> {
+    fn drop(&mut self) {
```
Comment (bilelmoussaoui) [inline glib/src/main_context.rs:145]: Could use a doc alias to help people finding out about the Drop implementation maybe?

Comment (sdroege) [reply glib/src/main_context.rs:145]: Doc alias for which function?

Comment (bilelmoussaoui) [reply glib/src/main_context.rs:145]: for drop, this way the user can find it by searching for the C function

Comment (sdroege) [reply glib/src/main_context.rs:145]: For `g_main_context_release` you mean?

Comment (bilelmoussaoui) [reply glib/src/main_context.rs:145]: Yes

Comment (sdroege) [reply glib/src/main_context.rs:146]: Ok, no idea if that actually has any effect though. Probably not :)

Comment (bilelmoussaoui) [reply glib/src/main_context.rs:146]: It does https://gtk-rs.org/gtk4-rs/git/docs/?search=gdk_event_get_type :)

File: glib/src/main_context.rs [hunk 2]
```diff
@@ -124,6 +124,30 @@ impl MainContext {
...
+            }
+        }
+    }
+}
+
+#[must_use = "if unused the main context will be released immediately"]
+pub struct MainContextAcquireGuard<'a>(&'a MainContext);
+
+impl<'a> Drop for MainContextAcquireGuard<'a> {
+    #[doc(alias = "g_main_context_acquire")]
```
Comment (bilelmoussaoui) [inline glib/src/main_context.rs:145]:
```suggestion
    #[doc(alias = "g_main_context_release")]
```

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**